### PR TITLE
Add force_kernel_name option

### LINKED ISSUE #190: ERR jupyter_client.kernelspec.NoSuchKernel: No such kernel named XXX
Ran into this issue by creating a notebook in one environment and then trying to run in a different environment jupyter gateway with notebook-http personality. Both environments are running Python 3.5.2

The issue comes from jupytergateway.py, which gets the kernel name from the .ipynb file:

`kernel_name = notebook['metadata']['kernelspec']['name']`

Creating the notebook in this first environment, which I had installed Python 3 using [these instructions](https://www.continuum.io/blog/developer-blog/python-3-support-anaconda) creates a .ipynb file with this metadata:

```
 "metadata": {
  "kernelspec": {
   "display_name": "Python [py35]",
   "language": "python",
   "name": "Python [py35]"
```

However the second environment doesn't have a kernel called "Python [py35]" thus the gateway command

`jupyter-kernelgateway --KernelGatewayApp.api=notebook-http\
                                     --KernelGatewayApp.seed_uri='test35.ipynb'`

 fails with:

```
ERR   File "/home/vcap/app/.conda/envs/dep_env/lib/python3.5/site-packages/kernel_gateway/gatewayapp.py", line 248, in _load_notebook
ERR     self.kernel_spec_manager.get_kernel_spec(kernel_name)
ERR   File "/home/vcap/app/.conda/envs/dep_env/lib/python3.5/site-packages/jupyter_client/kernelspec.py", line 175, in get_kernel_spec
ERR     raise NoSuchKernel(kernel_name)
ERR jupyter_client.kernelspec.NoSuchKernel: No such kernel named Python [py35]
```

The Python 3 kernel in the second environment is named "python3". I can get the code to work if I edit the metadata of the .ipynb file and change the name to "python3", though this seems not ideal. A potential solution would be if there was an option to force evaluation using a specified kernel name. E.g.,

-- KernelGatewayApp.kernel_name = "python3"

### PR DESCRIPTION
**Pull request by parente**
Override kernel names in requests and seed notebooks with one
specified at KG startup time

Fixes #190

### CODE REVIEW

**Comment by parente** [comment]
@crawles will this PR satisfy your use case?

**Comment by crawles** [comment]
@parente Yep, looks good to me. Nice!

**Comment by parente** [comment]
Cool. 

@nitind @jtyberg or anyone with merge: can you have a look? I'd like to cut a 1.1 release and with this in it.

File: kernel_gateway/gatewayapp.py [hunk 1]
```diff
@@ -250,8 +257,8 @@ def _load_notebook(self, uri):
 
         Raises
         ------
-        RuntimeError if no installed kernel can handle the language specified
-        in the notebook.
+        RuntimeError if there is no kernel speci matching the one specified in
```
Comment (jtyberg) [inline kernel_gateway/gatewayapp.py]: typo: `speci`

Comment (parente) [inline kernel_gateway/gatewayapp.py]: Fixed.

**Pull request merged**